pyro/contrib/minipyro.py

```diff
@@ -32,16 +32,19 @@ def get_param_store():
     return PARAM_STORE
 
 
+# The base effect handler class (called Messenger here for consistency with Pyro).
 class Messenger(object):
     def __init__(self, fn=None):
         self.fn = fn
 
+    # Effect handlers push themselves onto the PYRO_STACK.
+    # Handlers earlier in the PYRO_STACK are applied first.
     def __enter__(self):
-        PYRO_STACK.append(self)
+        PYRO_STACK.insert(0, self)
```

Comment:
IMO it's really gross to use Python's native stack datatype in reverse order of the intended use (as indicated by the `.pop()` method). This use is also slower due to linear-time cost of popping from the front, vs constant time cost of popping from the back. I think we should instead update full pyro to match this file.

Reply:
Sure, I just wanted `minipyro` to be consistent with actual Pyro.